lazy_tensor_core/lazy_tensor_core/core/lazy_tensor.py

```diff
@@ -0,0 +1,112 @@
+import torch
+import os
+import math
+ 
+class LazyDimension():
+ 
+    @staticmethod
+    def from_tensor(input, dim):
+        # TODO: not sure either _dim or _input should be exposed at all
+        # ideally we only need self._node
+ 
+        # self._dim = dim
+        # self._input = input
+ 
+        ld = LazyDimension()
+        ld._node = lazy_tensor_core._LAZYC._dynamic_size(input, dim)
+        return ld
+ 
+    @staticmethod
+    def from_node(node):
+      ld = LazyDimension()
+      ld._node = node
+      return ld  
+   
+    def __init__(self):
+        import inspect
+        import warnings
+        caller_name = inspect.stack()[1][3]
+        if caller_name not in ["from_node", "from_tensor"]:
+            warnings.warn("LazyDimension should only be called by \
+            `from_node` or `from_tensor`. This warning may be spurious \
+            if not using CPython")
+        pass
+       
+    @property
+    def dynamic(self):
+        lazy_tensor_core._LAZYC._is_dynamic(self._node)
+ 
+    @property
+    def static_value(self):
+        lazy_tensor_core._LAZYC._get_static_value(self._node)
+
+    def __len__(self, dim):
+        n = lazy_tensor_core._LAZYC._dynamic_size(self._node, dim)
+        return LazyDimension.from_node(n)
+ 
+    def __mul__(self, other):
+        assert (isinstance(other, LazyDimension))
+        n = lazy_tensor_core._LAZYC._dynamic_mul(self._node, other)
+        return LazyDimension.from_node(n)
+ 
+    def __div__(self, other):
+        assert (isinstance(other, LazyDimension))
+        n = lazy_tensor_core._LAZYC._dynamic_div(self._node, self.other)
+        return LazyDimension.from_node(n)
+
+    def __add__(self, other):
+        assert (isinstance(other, LazyDimension))
+        n = lazy_tensor_core._LAZYC._dynamic_add(self._node, self.other)
+        return LazyDimension.from_node(n)
+
+    # def __int__(self):
+    #     if not self.dynamic:
+    #         return self.static_value
+    #     else:
+    #         # TODO: if static_value can recursively fetch updated value
+    #         # and we can detect that tensors in leaves (i.e. aten::size) were materialized
+    #         # we can return the correct answer here
+    #         n = lazy_tensor_core._LAZYC._dynamic_int(self._node)
+    #         return LazyDimension.from_node(n)
+       
+    def __str__(self) -> str:
+        return f"LazyDimension({self._dim})"
+ 
+    def __repr__(self):
+        return str(self)
+ 
+class LazyTensor(torch.Tensor):
```

Comment:
@Krovatkin, do we still need to keep the `LazyTensor` class in this implementation? If so, (a) how does it work with `BoxedInt`, (b) does it need to subclass from `torch.Tensor`?